src/pointer.rs

```diff
@@ -0,0 +1,88 @@
+//! Abstractions over raw pointers.
+
+mod ptr;
+
+pub use maybe_aligned::MaybeAligned;
+pub use ptr::{invariant, Ptr};
+
+/// A shorthand for a maybe-valid, maybe-aligned reference. Used as the argument
+/// to [`TryFromBytes::is_bit_valid`][crate::TryFromBytes::is_bit_valid].
+pub type MaybeValid<'a, T> =
```

Comment:
Can we call this `MaybeAlignedValid` or something? I was confused by the naming when reading `lib.rs` until I came and looked at this doc comment.

Reply:
Renamed to `Maybe<T>`. >:]

I'm signing off for the night, but feel free to bikeshed the name without me!

Reply:
Can we add a separate `#[doc(hidden)]` attribute here (so that it remains even if we remove the outer one on the module) and leave a TODO comment to bikeshed this name if we're planning on making this public?